app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -173,15 +173,7 @@ def save_hosts_inventory(ems, hashes, target = nil, disconnect = true)
           found.update_attributes(h)
         end
 
-        # Handle duplicate names coming in because of duplicate hostnames.
-        begin
-          found.save!
-        rescue ActiveRecord::RecordInvalid
-          raise if found.errors[:name].blank?
-          old_name = Host.where("name LIKE ?", "#{found.name.sub(/ - \d+$/, "")}%").order("LENGTH(name) DESC").order("name DESC").first.name
-          found.name = old_name =~ / - \d+$/ ? old_name.succ : "#{old_name} - 2"
-          retry
-        end
```

Comment:
This still could happen if someone doesn't set the hostname in vsphere, so they all end up being localhost.localdomain...  However, if you get rid of the uniqueness check, then I guess this code doesn't matter.

Reply:
Not sure if this will help but the changes were in response to https://bugzilla.redhat.com/show_bug.cgi?id=942903.  

Commit cda24a3372 comment on Nov 2011:
> Changed EMS Refresh to handle duplicate names coming in because of duplicate hostnames by appending a number to the name.

Reply:
Right it was failing because of the uniqueness validation, I tried to find the history on that but it was in the initial import.

Without the unique validation this code wouldn't be needed.